File: app/src/main/java/com/example/sqlite/db/DbContactos.java
package com.example.sqlite.db;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import androidx.annotation.Nullable;

import com.example.sqlite.entidades.Contactos;

import java.util.ArrayList;

public class DbContactos extends DbHelper{

    Context context;

    public DbContactos(@Nullable Context context) {
        super(context);
        this.context = context;
    }

    public long insertarContacto(String nombre,String telefono, String email){
        long id = 0;
        try {
            DbHelper dbHelper = new DbHelper(context);
            SQLiteDatabase db = dbHelper.getWritableDatabase();

            ContentValues values = new ContentValues();
            values.put("nombre", nombre);
            values.put("telefono", telefono);
            values.put("email", email);

            id = db.insert(TABLE_CONTACTOS, null, values);
        }catch (Exception ex){
            ex.toString();
        }
        return id;
    }

    public ArrayList<Contactos> mostrarContactos(){
        DbHelper dbHelper = new DbHelper(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();

        ArrayList<Contactos> listaContactos = new ArrayList<>();
        Contactos contacto = null;
        Cursor cursorContactos = null;

        cursorContactos = db.rawQuery("SELECT * FROM " + TABLE_CONTACTOS, null);

        if(cursorContactos.moveToFirst()){
            do{
                contacto = new Contactos();
                contacto.setId(cursorContactos.getInt(0));
                contacto.setNombre(cursorContactos.getString(1));
                contacto.setTelefono(cursorContactos.getString(2));
                contacto.setEmail(cursorContactos.getString(3));
                listaContactos.add(contacto);
            }while (cursorContactos.moveToNext());
        }

        cursorContactos.close();

        return  listaContactos;
    }

}

File: app/src/main/java/com/example/sqlite/db/DbContactos.kt
package com.example.sqlite.db

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import com.example.sqlite.entidades.Contactos

class DbContactos(context: Context?) : DbHelper(context) {

    private val context: Context? = context

    fun insertarContacto(nombre: String, telefono: String, email: String): Long {
        var id: Long = 0
        try {
            val dbHelper = DbHelper(context)
            val db = dbHelper.writableDatabase

            val values = ContentValues().apply {
                put("nombre", nombre)
                put("telefono", telefono)
                put("email", email)
            }

            id = db.insert(TABLE_CONTACTOS, null, values)
        } catch (ex: Exception) {
            ex.toString()
        }
        return id
    }

    fun mostrarContactos(): ArrayList<Contactos> {
        val dbHelper = DbHelper(context)
        val db = dbHelper.writableDatabase
        val listaContactos = ArrayList<Contactos>()
        var contacto: Contactos? = null
        var cursorContactos: Cursor? = null
        cursorContactos = db.rawQuery("SELECT * FROM " + TABLE_CONTACTOS, null)
        if (cursorContactos.moveToFirst()) {
            do {
                contacto = Contactos()
                contacto.id = cursorContactos.getInt(0)
                contacto.nombre = cursorContactos.getString(1)
                contacto.telefono = cursorContactos.getString(2)
                contacto.email = cursorContactos.getString(3)
                listaContactos.add(contacto)
            } while (cursorContactos.moveToNext())
        }
        cursorContactos.close()
        return listaContactos
    }

}